### PR TITLE
Scaffolded app and set up tests

### PR DESCRIPTION
**Pull request by larathordar**
#### Description
Changes proposed in this pull request:

* rails app scaffolded
* gem factory_bot_rails, gem pry-byebug', gem 'pry-rails', gem 'rspec-rails', gem 'shoulda-matchers' and chrome helper added to gemfile

* Testing environment for rspec and cucumber implemented

#### What I have learned working on this feature: 
Every single user story needs a separate git branch. Learned to implement chrome helper.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  gem 'spring'
+  gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'factory_bot_rails'
+  gem 'pry-byebug'
+  gem 'pry-rails'
+  gem 'rspec-rails'
+  gem 'shoulda-matchers'
+end
+
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```
Comment (magnus-thor) [inline Gemfile]: remove this line, its only needed if you are working on windows

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+group :development do
+  gem 'web-console', '>= 3.3.0'
+  gem 'listen', '>= 3.0.5', '< 3.2'
+  gem 'spring'
+  gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'factory_bot_rails'
+  gem 'pry-byebug'
+  gem 'pry-rails'
+  gem 'rspec-rails'
+  gem 'shoulda-matchers'
```
Comment (magnus-thor) [inline Gemfile]: move shoulda-matchers, rspec-rails, pry-byebug, and pry-rails and factory_bot_rails to `group :development, :test do
`

File: app/jobs/application_job.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class ApplicationJob < ActiveJob::Base
+end
```
Comment (magnus-thor) [inline app/jobs/application_job.rb]: You can remove this file

File: app/assets/javascripts/cable.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
+
+}).call(this);
```
Comment (magnus-thor) [inline app/assets/javascripts/cable.js]: you can remove this file.

File: app/channels/application_cable/channel.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Channel < ActionCable::Channel::Base
+  end
+end
```
Comment (magnus-thor) [inline app/channels/application_cable/channel.rb]: remove this file

File: app/channels/application_cable/connection.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+  end
+end
```
Comment (magnus-thor) [inline app/channels/application_cable/connection.rb]: remove this file

File: config/cable.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
+  channel_prefix: euphony_production
```
Comment (magnus-thor) [inline config/cable.yml]: remove this file

File: config/database.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  timeout: 5000
+
+development:
+  <<: *default
+  database: db/development.sqlite3
+
+# Warning: The database defined as "test" will be erased and
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
```
Comment (magnus-thor) [inline config/database.yml]: remove commented out lines

File: config/environments/development.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  config.assets.debug = true
+
+  # Suppress logger output for asset requests.
+  config.assets.quiet = true
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
+
+  # Use an evented file watcher to asynchronously detect changes in source code,
+  # routes, locales, etc. This feature depends on the listen gem.
```
Comment (magnus-thor) [inline config/environments/development.rb]: remove comments

File: config/environments/production.rb [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  # require 'syslog/logger'
+  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
+
+  if ENV["RAILS_LOG_TO_STDOUT"].present?
+    logger           = ActiveSupport::Logger.new(STDOUT)
+    logger.formatter = config.log_formatter
+    config.logger    = ActiveSupport::TaggedLogging.new(logger)
+  end
+
+  # Do not dump schema after migrations.
```
Comment (magnus-thor) [inline config/environments/production.rb]: remove comments

File: config/environments/test.rb [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  # Tell Action Mailer not to deliver emails to the real world.
+  # The :test delivery method accumulates sent emails in the
+  # ActionMailer::Base.deliveries array.
+  config.action_mailer.delivery_method = :test
+
+  # Print deprecation notices to the stderr.
+  config.active_support.deprecation = :stderr
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
```
Comment (magnus-thor) [inline config/environments/test.rb]: remove comments

File: features/support/env.rb [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+#   end
+#
+#   Before('~@no-txn', '~@selenium', '~@culerity', '~@celerity', '~@javascript') do
+#     DatabaseCleaner.strategy = :transaction
+#   end
+#
+
+# Possible values are :truncation and :transaction
+# The :transaction strategy is faster, but might give you threading problems.
+# See https://github.com/cucumber/cucumber-rails/blob/master/features/choose_javascript_database_strategy.feature
```
Comment (magnus-thor) [inline features/support/env.rb]: remove comments

**Review by magnus-thor — COMMENTED** [review]
Remember one story per pull request.

File: app/channels/application_cable/connection.rb (hunk 6 above)
Comment (magnus-thor) [reply app/channels/application_cable/connection.rb]: remove this file

File: config/cable.yml (hunk 7 above)
Comment (magnus-thor) [reply config/cable.yml]: remove this file

File: config/environments/development.rb (hunk 9 above)
Comment (magnus-thor) [reply config/environments/development.rb]: clean out the comments

File: Gemfile [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+  repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
+  "https://github.com/#{repo_name}.git"
+end
+
+gem 'rails', '~> 5.1.5'
+gem 'pg'
+gem 'puma', '~> 3.7'
+gem 'sass-rails', '~> 5.0'
+gem 'uglifier', '>= 1.3.0'
+gem 'coffee-rails', '~> 4.2'
```
Comment (tochman) [inline Gemfile]: Remove this gem

File: Gemfile [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+gem 'turbolinks', '~> 5'
+gem 'jbuilder', '~> 2.5'
+
```
Comment (tochman) [inline Gemfile]: Remove this gem

File: Gemfile [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+group :development, :test do
+  gem 'cucumber-rails', require: false
+  gem 'database_cleaner'
+  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
```
Comment (tochman) [inline Gemfile]: Remove this gem

**Review by tochman — CHANGES_REQUESTED** [review]

**Review by tochman — APPROVED** [review]

**Pull request merged**